bokeh/tests/test_events.py

```diff
@@ -22,11 +22,11 @@ def __call__(self, event):
 
 def test_event_metaclass():
     # All events currently in the namespace should be in the EVENT_CLASSES set
-    assert len(all_events - events.EVENT_CLASSES) == 0
+    assert len(concrete_events - set(events._CONCRETE_EVENT_CLASSES.values())) == 0
 
 def test_common_decode_json():
-    for event_cls in events.EVENT_CLASSES:
-        if event_cls.event_name is None: continue # Skip abstract base class
+    for event_name, event_cls in events._CONCRETE_EVENT_CLASSES.items():
```

Comment:
It's generally not recommended to loop over iterables inside of pytest tests. The suggested thing to do is use the `@pytest.mark.parametrize` decorator. Idk if it's worth messing with now though